### ISSUE TITLE
Consider releasing a non-beta version after transition to 64-bit is complete

### ISSUE THREAD

**Issue by dondi** [lmu-bioinformatics/xmlpipedb#5]
While we are thinking about version naming and such, in doing this most recent wave of work the thought occurred to me that it may be a good time to release something without the beta label. We can either go with a final 2.0 version, or rename the last relatively stable pre-64-bit version as 2.0 (was that b84? or b83?) and move to 3.0 with this version that is now pure Java, including the export functionality.

Open to any thoughts on this...

**Comment by kdahlquist** [comment]
Here are my thoughts:

* I'm not so keen on retro-actively labeling something 2.0.  I think we should just go ahead and increment to 3 and adopt the 3.x.x convention for naming releases without altering any of our previous stuff.
* I think what had been holding us back previously from removing the "beta" designation was that we were going to run through all of the different species exports to see if the respective species profiles that we haven't touched in a long time still work.  However, I don't have time for this either.  So I'm back to thinking that we should just go ahead and increment to 3.

**Comment by dondi** [comment]
OK, when this particular version is finished we will call it 3.0b1 then. The bump to 3 communicates the significant shift under the hood. In a sense it also cleanly delineates our going to Java 8. We will retain the "b1" to show that existing species profiles need testing. This is actually especially true now with all of the export engine changes.

**Comment by kdahlquist** [comment]
I was thinking of actually getting rid of "beta" entirely from the name.  I guess it's true that we have a beta branch and version of GRNsight, but we number it without using beta in the name.

I think we should do the same here.  Branch off beta for development and then merge it back to master and release.  It will be a looooooooong time before we can verify all the species profiles and I don't want to be stuck with the beta designation forever.  If there are bugs with them, then we'll just open issues and then fix them.  Is that OK?

Also, let's adopt our naming convention of using even numbers for released versions and odd numbers for development versions like we do with GRNsight.

I know this is a different convention than we have used historically with XMLPipeDB, but we have an opportunity now to bring it up to the best practices we have adopted for the other projects.

**Comment by dondi** [comment]
Ah OK, no problem with removing our prior “beta” constraint. This can be 3.0 flat out then :) And we will fix as we go. The existing `access-64-bit` branch can be considered as the beta branch leading up to this release.

Taking this a step further into current practices, in actuality branching into `beta` was actually a simplification to get students into the habit of working on a separate branch while something is in progress. The full-blown model is that everyone works on a different branch whose name represents the specific task on that branch. Then, as those tasks complete, they are merged into master. Version numbers are determined based on which concurrent tasks “graduate to master” first.

I'm not sure if we are ready as a group to go into that—for this current work, because it involves just me, I did follow that convention more closely. For the impending class, one effect of switching to the full branch process is that each group would actually have their own branch, named after the species customization that they are doing. They are then able to work in isolation until their specific species is ready.

But anyway, that is going beyond the scope of this particular issue somewhat. We'll conclude this one by saying that we will go to GenMAPP Builder 3.0 for the current release-in-progress.

**Comment by kdahlquist** [comment]
Got it about the naming of the branches.  For sure when we have the teams we should operate that way.  I was also thinking of creating labels for the guilds and teams (labeled by species so that they are readable).  I will close, since we have settled on an answer.
